DNN Platform/DotNetNuke.Web/Api/Auth/AuthMessageHandlerBase.cs

```diff
@@ -60,6 +107,11 @@ protected AuthMessageHandlerBase(bool includeByDefault, bool forceSsl)
         /// <returns>null normally, if a response is returned all inbound processing is terminated and the resposne is returned.</returns>
         public virtual HttpResponseMessage OnInboundRequest(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (request.Headers.Contains("Origin"))
+            {
+                this.origin = request.Headers.GetValues("Origin").FirstOrDefault();

```

Comment:
Is there one instance of this `DelegatingHandler` per request?  Do we know it's safe to store this into in the object instance, or could it be reused by multiple requests?

Reply:
Excellent question, this will need to be re-taught and well tested. I kinda got busy and could not get back to this. So I put it on hold but this still needs some work... I'll try to revisit this again this weekend or some such...